src/Util/Logger/FriendicaProcessor.php

```diff
@@ -47,13 +47,24 @@ public function __invoke(array $record)
 
 		$i = 1;
 
-		// Skip everything that we shouldn't display
-		while (in_array($trace[$i]['function'], $this->skipFunctions) ||
-			!isset($trace[$i - 1]['file']) ||
-			in_array(basename($trace[$i - 1]['file']), $this->skipFiles)) {
-			$i++;
+		while ($this->isTraceClassOrSkippedFunction($trace, $i)) {
```

Comment:
I still don't understand the need for the complexity of this construct. We only have to repeat this structure while we are inside the levels that we want to skip. So this in fact could be some ```while (true)``` or - better - we could transform this into some simple ```do ... while``` loop (since we know that the first levels will always have to be skipped).